libs/model/src/policies/CreateUnverifiedUser.policy.ts

```diff
@@ -0,0 +1,58 @@
+import { Policy } from '@hicommonwealth/core';
+import { events } from '@hicommonwealth/schemas';
+import { Op } from 'sequelize';
+import { ZodUndefined } from 'zod';
+import { models } from '../database';
+
+const inputs = {
+  TokenLocked: events.TokenLocked,
+};
+
+/**
+ * Creates an unverified common user when a new TokenLocked chain event is received and user doesn't exists
+ */
+export function CreateUnverifiedUser(): Policy<typeof inputs, ZodUndefined> {
+  return {
+    inputs,
+    body: {
+      TokenLocked: async ({ payload }) => {
+        const { address } = payload.parsedArgs;
+
+        await models.sequelize.transaction(async (transaction) => {
+          // address already linked to a user
+          const found = await models.Address.findOne({
+            where: { address, user_id: { [Op.ne]: null } },
+            attributes: ['community_id', 'user_id'],
+            transaction,
+          });
+          if (found) return;
+
+          const user = await models.User.create(
+            {
+              email: null,
+              profile: {
+                name: address.substring(0, 8), // TODO: unclaimed name
```

Comment:
@sachben91 & @dillchen : do we create a default profile for these users?

Reply:
Yea it would be good to do so

Can likely leverage:

import { generateUsername } from 'unique-username-generator';

https://github.com/hicommonwealth/commonwealth/blob/ebed75c7fad94da216d13f9197e68c9913432abc/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx#L18